[Ansel Chong] iP

Ansel-Ch opened the pull request:
## Joe
> “The same boiling water that softens the potato hardens the egg. It's about what you're made of, not the circumstances.” - Roald Dahl ([source](https://www.goodreads.com/quotes/11675966-the-same-boiling-water-that-softens-the-potato-hardens-the))

Joe frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
All you need to do is,
1. download it from [here](https://github.com/Ansel-Ch/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
___
If you are a Java programmer, you can use it to practice Java too. Here's the <span style="color:#ff69b4;">main</span> method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Ansel-Ch committed:
Rename Duke class to Joe class

Ansel-Ch committed:
Add greeting

Ansel-Ch committed:
Add echo method and bye method

Ansel-Ch committed:
Update to echo to include joe's responeses

Ansel-Ch committed:
Add add and list

Ansel-Ch committed:
Add Task class

Ansel-Ch committed:
Update Task class

Ansel-Ch committed:
Update to add and list methods to support mark functionality

Ansel-Ch committed:
Update to toString and setDone methods in Task class

Ansel-Ch committed:
Added mark method and renamed static variables

Ansel-Ch committed:
Add unmark method

Ansel-Ch committed:
Added ToDo class and updated Task class

Ansel-Ch committed:
Add Deadline and Event class

Ansel-Ch committed:
Add support for tracking ToDos, Events and Deadlines

Ansel-Ch committed:
Update EXPECTED.TXT and input.txt for UiTesting

Ansel-Ch committed:
Fix bug in Deadline

Ansel-Ch committed:
Bug fix for Event

Ansel-Ch committed:
Bug fix in ToDo constructor

Ansel-Ch committed:
Bug fix in list and unmark

Ansel-Ch committed:
Add IllegalTaskException and InvalidCommandException

Ansel-Ch committed:
Update Task constructor to throw empty descriptions

Ansel-Ch committed:
Add help function to support InvalidCommandException

Ansel-Ch committed:
Delete IllegalTaskException

Ansel-Ch committed:
Update Deadline constructor to catch No Due Date inputs

Ansel-Ch committed:
Update Event constructor is catch illegal arguments

Ansel-Ch committed:
Update to Task constructor after deletion of IllegalTaskException

Ansel-Ch committed:
Update add method to catch new Exceptions

Ansel-Ch committed:
Add support for delete by changing array to arraylist

Ansel-Ch committed:
Update error message for invalid commands

Ansel-Ch committed:
Add delete and Update help method to reflect new command

Ansel-Ch committed:
Document all classes and methods

Ansel-Ch committed:
Update toString methods

Ansel-Ch committed:
Update to tests for new exception handling and help function

Ansel-Ch committed:
Update for case-sensitivity in Joe class

Ansel-Ch committed:
Update Deadline constructor

Ansel-Ch committed:
Update Event Constructor

Ansel-Ch committed:
Add Exception handling in add function

Ansel-Ch committed:
Update documentation for all java files

Ansel-Ch committed:
Update to documents in Joe.java and Task.java

Ansel-Ch committed:
Add joe.txt

Ansel-Ch committed:
joe.txt: Add test task entries

Ansel-Ch committed:
Add CorruptedFileException

Ansel-Ch committed:
CorruptedFileException.java: Add constructor

Ansel-Ch committed:
Joe.java: Add syncdata method to load tasks from joe.txt

Ansel-Ch committed:
Event.java: Add getStart and getEnd methods

Ansel-Ch committed:
Deadline.java: Add getDue method

Ansel-Ch committed:
joe.txt: Amend "June 6th" to "by June 6th"

Ansel-Ch committed:
Joe.java: Add save method

Ansel-Ch committed:
Joe.java: Remove .toLowerCase() from userIn

Ansel-Ch committed:
Merge branch 'branch-Level-7'

Ansel-Ch committed:
Deadline.java: Update due field to LocalDateTime variable

Ansel-Ch committed:
joe.txt & input.txt: Update test input to meet parsing format

Ansel-Ch committed:
Event.java: Update start and end fields to LocalDateTime

Ansel-Ch committed:
Deadline.java: Update toString method

Ansel-Ch committed:
joe.txt: Change string format for event to match new Event parsing

Ansel-Ch committed:
Add support for Deadline and Event LocalDateTime changes

Ansel-Ch committed:
Add support for query method in Joe.java

Ansel-Ch committed:
Joe.java: Add queryTasksByDate method

Ansel-Ch committed:
Joe.java: Add documentation for syncData and save methods

Ansel-Ch committed:
Merge branch 'branch-Level-8'

Ansel-Ch committed:
Add Formatter.java, InvalidIndexException.java, Parser.java, Storage.java, TaskList.java, Ui.java

Ansel-Ch committed:
Deadline.java, Event.java: Refactor code to use Formatter class methods

Ansel-Ch committed:
InvalidCommandException: Change overridden method to getMessage method

Ansel-Ch committed:
Joe.txt: Refactor code to use new java classes

Ansel-Ch committed:
Update test files to match new formatting

Ansel-Ch committed:
Add javadocs for getters in Deadline.java and Event.java

Ansel-Ch committed:
Merge branch 'branch-A-MoreOOP'

Ansel-Ch committed:
Remove all getters and add new instance methods to support those operations using the getters

Ansel-Ch committed:
Formatter.java: Remove redundant task formatters methods

Ansel-Ch committed:
TaskList.java: Update to use new instance methods for Deadline and Event class

Ansel-Ch committed:
Package all java files to respective packages

Ansel-Ch committed:
Merge branch 'add-gradle-support'

Ansel-Ch committed:
Rename package to joe.Main

Ansel-Ch committed:
build.gradle: Configure file to match package names

Ansel-Ch committed:
Add CorruptedFileExceptionTest.java

Ansel-Ch committed:
Add InvalidCommandExceptionTest.java

Ansel-Ch committed:
Add InvalidIndexExceptionTest.java

Ansel-Ch committed:
Add test.txt for unit testing

Ansel-Ch committed:
Add StorageTest.java

Ansel-Ch committed:
Joe.java: Add initializeTask method for unit testing

Ansel-Ch committed:
Parser.java: Fix bug where corrupted file will not produce enough parameters and raise ArrayIndexOutOfBoundsException

Ansel-Ch committed:
Add getters for unit testing

Ansel-Ch committed:
Add corrupted.txt to produce CorruptedFileException

Ansel-Ch committed:
Add JoeTest.java

Ansel-Ch committed:
Add ParserTest.java

Ansel-Ch committed:
Merge branch 'branch-A-JUnit'

Ansel-Ch committed:
build.gradle: Update archieveBaseName for jar

Ansel-Ch committed:
Add javadocs for the remaining classes

Ansel-Ch committed:
Move txt files to root folder data directory

Ansel-Ch committed:
Task.java: Change to abstract class

Ansel-Ch committed:
TaskList.java: Add find method

Ansel-Ch committed:
Parser.java: Add support for find command

Ansel-Ch committed:
Joe.java: Remove redundant comment

Ansel-Ch committed:
TaskList.java: Add javadocs for find method

Ansel-Ch committed:
Merge branch 'branch-A-JavaDoc'

Ansel-Ch committed:
Ui.java: Add line for use of find command

Ansel-Ch committed:
Merge branch 'branch-A-CodingStandard'

Ansel-Ch committed:
Add abstract Command class and its respective subclasses

Ansel-Ch committed:
TaskList.java: Update methods to support Command subclasses

Ansel-Ch committed:
Ui.java: Update methods to make use of Command classes

Ansel-Ch committed:
Parser.java: Update parseCommand method to pass command back to UI to handle response

Ansel-Ch committed:
joe.txt: Change to line due to save testing

Ansel-Ch committed:
Merge branch 'branch-Level-9'

Ansel-Ch committed:
TaskList.java: Add header comments

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Implement checkstyle

Ansel-Ch committed:
Merge branch 'branch-A-CheckStyle'

Ansel-Ch committed:
Task.java: Update code

aditig0305 commented on src/main/java/joe/Main/Parser.java:
```diff
@@ -0,0 +1,123 @@
+package joe.Main;
```
```suggestion
package joe.main;
```
Should the name for package be in all lower case?

Ansel-Ch replied:
Yea you are right.

aditig0305 commented on src/main/java/joe/Main/Parser.java:
```diff
@@ -0,0 +1,123 @@
+     * @param userCmd the String representation of the user input
+     * @return a Command object representing the user input
+     */
+    public Command parseCommand(String userCmd) throws IllegalArgumentException, InvalidCommandException {
```
I am impressed by each type of input having its own detailed command object. Nice!

aditig0305 commented on src/main/java/joe/Main/Storage.java:
```diff
@@ -0,0 +1,43 @@
+ */
+public class Storage {
+
+    private static final String line =
```
If this is a constant should it be all capitalised?
```suggestion
    private static final String LINE =
```

Ansel-Ch replied:
Yeap you are right, they should be in screaming snake_case. Thanks for pointing it out 😄

aditig0305 commented on src/main/java/joe/Main/Parser.java:
```diff
@@ -0,0 +1,123 @@
+     * @param userCmd the String representation of the user input
+     * @return the numerical substring as an integer
+     */
+    public int getDigits(String userCmd) {
```
Method names are following the guidelines. Nice

aditig0305 commented on src/main/java/joe/Main/Joe.java:
```diff
@@ -0,0 +1,64 @@
+
+/**
+ * Represents the main class of the programme.
+ */
```
Nice that Javadoc comments are seen in the files. Consistent :)

aditig0305 commented on src/test/java/joe/Main/ParserTest.java:
```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Tests the parseTask method in Parser.
+     */
```
Detailed javadoc comments for each detailed test case. Nice

aditig0305 commented on src/main/java/joe/Main/Parser.java:
```diff
@@ -0,0 +1,123 @@
+     * @return a Command object representing the user input
+     */
+    public Command parseCommand(String userCmd) throws IllegalArgumentException, InvalidCommandException {
+        Command c;
```
Since the variable c is quite important here should it perhaps have a more intuitive name?

aditig0305 commented on src/main/java/joe/tasks/TaskList.java:
```diff
@@ -0,0 +1,211 @@
+            FileWriter fw = new FileWriter("data/joe.txt");
+            fw.write(sb.toString());
+            fw.close();
+            System.out.println("Your tasks have been successfully saved.");
```
Maybe for methods or chunks of codes such as these, there can be lines in between to help the reader to understand how the code logic flows.

aditig0305 commented on src/main/java/joe/tasks/Task.java:
```diff
@@ -0,0 +1,73 @@
+     *
+     * @param done a boolean that determines whether a task is done
+     */
+    public void setDone(boolean done) {
```
I think the boolean name has to follow the naming standard.
```suggestion
    public void setDone(boolean isDone) {
```

Ansel-Ch replied:
ooo, yea i must have missed out on that parameter. Thanks

aditig0305 reviewed (commented):
Overall most coding standards are met. I did not have many difficulties while trying to read the code.

Ansel-Ch committed:
Implement GUI

Ansel-Ch committed:
Merge branch 'branch-Level-10'

Ansel-Ch committed:
Update Gui

Ansel-Ch committed:
Update gui layout

Ansel-Ch committed:
Fix delete command bug

Ansel-Ch committed:
Add assertions

In the Parser class,  it should be checked whether the arguments
passed into certain commands follow the strict formatting of the
command option.

Formatter was also removed and static fields and methods
are moved to Parser.

This was a step towards removing redundant code to improve
code quality.

Ansel-Ch committed:
Enhance Code Quality

The main bulk of arrowhead code in the Parser parseCommand
method was enhanced to using the SLAP principle. Several
private static methods were extracted for this purpose.

A bug was also fixed in the getTaskIndex method where user not
inputting an index was not accounted for. The new guard
statement will raise an InvalidCommandException when this
specific user error is made.

Ansel-Ch committed:
Merge pull request #1 from Ansel-Ch/branch-A-Assertions

Add assertions

Ansel-Ch committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/joe/utils/Parser.java

Ansel-Ch committed:
Merge pull request #2 from Ansel-Ch/branch-A-CodeQuality

Enhance Code Quality

Ansel-Ch committed:
Implement streams

Ansel-Ch committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/joe/tasks/TaskList.java

Ansel-Ch committed:
Merge pull request #3 from Ansel-Ch/branch-A-Streams

Implement streams

Ansel-Ch committed:
Implement CI

Ansel-Ch committed:
Implement view schedule feature

Ansel-Ch committed:
Merge pull request #4 from Ansel-Ch/branch-BCD-Extension

Implement view schedule feature

Ansel-Ch committed:
Implement checking for duplicates in task adding

Ansel-Ch committed:
Fix bugs in Duplicate Extension

Ansel-Ch committed:
Enhance code quality by extracting long methods

Ansel-Ch committed:
Update GUI

Changes include new color changing depending on command
and images no longer have backgrounds.

Ansel-Ch committed:
Add Ui.png

Ansel-Ch committed:
Add user guide

Ansel-Ch committed:
Fix User guide command summary table

Ansel-Ch committed:
Update command summary

Ansel-Ch committed:
Fix table render bug

Ansel-Ch committed:
Fix table render

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update READMe.md

Ansel-Ch committed:
Merge branch 'master' of https://github.com/Ansel-Ch/ip

# Conflicts:
#	docs/README.md

Ansel-Ch committed:
Fix syntax for README

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Update README.md

Ansel-Ch committed:
Fix InvocationException bug

Ansel-Ch committed:
Merge branch 'master' of https://github.com/Ansel-Ch/ip
